### ISSUE TITLE
Change in behaviour for page resources since v0.56

### ISSUE THREAD

**Issue by stp-ip** [gohugoio/hugo#6208]
We are using `Resources.ByType` to aggregate all content files within subdirectories making up a site. This way splitting of content files is enabling a page being made up of various fragments.
Fragments located in [page]/[subdirectory]/index.md don't work since Hugo 0.56 doesn't recognize [subdirectory]/index.md as page resources anymore. Same for 0.57.
Current workaround is to create the fragment as [page]/fragment.md in addition to [subdirectory].
Basically moving index.md to ../fragment.md. That way it is recognized and the theme still works.

This feels like a regression. If this was a misusage, happy to get this documented and fixed on our side.

Specifically this came up with the [Syna](https://syna.okkur.org) Theme.
This is also breaking the demo: https://themes.gohugo.io//theme/syna/fragments/faq/

**Comment by bep** [comment]
It would be simpler if you could just point to a content directory showing this problem. I assume this is one example?:

https://github.com/okkur/syna/tree/master/exampleSite/content/fragments/faq
